### PR TITLE
Update/fixed versioning

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

This adds a `requirements.txt` file that can be used to specify the versions of **all** PIP packages required by F´. This allows users to install from a known, working, CI-verified working configuration.

Fixes: https://github.com/nasa/fprime/issues/1455

Note: for now, `setup.py` remains as it can be used for developers needing to update the `requirements.txt` file when versions need to be bumped.

### CODE REVIEW

File: docs/INSTALL.md [hunk 1]
```diff
@@ -49,7 +49,8 @@ tool installations are described in [Advanced](#Advanced).
 **Installing F´ Python Packages**
 
 ```
-pip install fprime/
+pip install -U setuptools wheel pip
+pip install -U -r fprime/requirements.txt
```
Comment (capsulecorplab) [inline docs/INSTALL.md:53]: Any particular reason to use the `-U` flag here? If a python dependency makes a breaking change, I feel like that kind of defeats the point of using a `requirements.txt` for reproducible builds.

Comment (capsulecorplab) [inline docs/INSTALL.md:53]: Any particular reason to use the `-U` flag here? If a python dependency makes a breaking change, I feel like that kind of defeats the point of using a `requirements.txt` for reproducible builds.

Comment (capsulecorplab) [reply docs/INSTALL.md:53]: I guess the `-U` flag is ignored if a specific version is specified pip install -U flag ignored if specific version is specified

Comment (LeStarch) [reply docs/INSTALL.md:53]: Yes, here we are using the -U flag to force the version to that specified.  The use case being: you've install a previous revision's requirements.txt and need to upgrade/downgrade.  The -U was intended to force a change from what you have to what is needed.

Perhaps this is not necessary. I will test.

**Review by timcanham — APPROVED** [review]
Looks good!

**Pull request merged**